src/costmap_to_polygons.cpp

```diff
@@ -52,52 +52,51 @@ namespace
  * 
  * Douglas-Peucker Algorithm, see https://en.wikipedia.org/wiki/Ramer%E2%80%93Douglas%E2%80%93Peucker_algorithm
  * 
- * @param pointList the initial set of points
- * @param epsilon distance criteria for removing point if it is smaller to the line segment than this
+ * @param begin iterator pointing to the begin of the range of points
+ * @param end interator pointing to the end of the range of points
+ * @param epsilon distance criteria for removing points if it is closer to the line segment than this
  * @param result the simplified polygon
  */
-void douglasPeucker(const std::vector<geometry_msgs::Point32> &pointList, double epsilon,
-                    std::vector<geometry_msgs::Point32> &result)
+std::vector<geometry_msgs::Point32> douglasPeucker(std::vector<geometry_msgs::Point32>::iterator begin,
+  std::vector<geometry_msgs::Point32>::iterator end, double epsilon)
 {
-  if (pointList.size() <= 2)
+  if (std::distance(begin, end) <= 2)
   {
-    result = pointList;
-    return;
+    return std::vector<geometry_msgs::Point32>(begin, end);
   }
 
-  // Find the point with the maximum distance from line between start and end
+  // Find the point with the maximum distance from the line [begin, end)
   double dmax = 0.0;
```

Comment:
nice commit. the last thing I have to object is dmax = 0.0. If it happens, that all points lie on a line, we would never init max_dist_it. I would init dmax with a negative value (like std::numeric_limits<T>::lowest)